### PR TITLE
Change license of node-template and FRAME examples to MIT-0

### PR DESCRIPTION
**Pull request by aaronbassett**
This PR changes the licensing of the node-template and FRAME examples from the [Unlicense](https://opensource.org/license/unlicense/) to [MIT No Attribution License](https://opensource.org/license/mit/).

While I ideologically agree with the Unlicense, licensing software with the Unlicense can have the opposite effect of the desired goal; it can in reality make it impossible for people to use the software freely.

For more information on why this is:

- [OSI Mailing list](https://web.archive.org/web/20170301020915/https://lists.opensource.org/pipermail/license-review/2012-January/001386.html)
- [What is wrong with the Unlicense?](https://softwareengineering.stackexchange.com/questions/147111/what-is-wrong-with-the-unlicense)

I chose the MIT No Attribution License (MIT-0) as the replacement, as this license seems to best align with the goals of the Unlicense while also being globally enforceable/valid.

## Why not CC0?

The CC0 license [potentially has the same issue as the Unlicense](https://wiki.creativecommons.org/wiki/CC0_FAQ#Does_CC0_really_eliminate_all_copyright_and_related_rights.2C_everywhere.3F)

## Additional Notes

The [fast-unstake](https://github.com/paritytech/substrate/blob/1837f423b494254e1d27834b1c9da34b2c0c2375/frame/fast-unstake/Cargo.toml#L6) and [transaction-storage](https://github.com/paritytech/substrate/blob/1837f423b494254e1d27834b1c9da34b2c0c2375/frame/transaction-storage/Cargo.toml#L6) pallets also use the Unlicense. I'll create a separate PR to update them to Apache-2 to keep them consistent with the other FRAME pallets.

### CODE REVIEW

**Comment by bkchr** [comment]
From your links it makes, but I'm no lawyer, so I asked them :P

**Review by bkchr — APPROVED** [review]
Lawyer says yes!

File: bin/node-template/LICENSE [hunk 1]
```diff
@@ -1,24 +1,16 @@
-This is free and unencumbered software released into the public domain.
+MIT No Attribution
 
-Anyone is free to copy, modify, publish, use, compile, sell, or
-distribute this software, either in source code form or as a compiled
-binary, for any purpose, commercial or non-commercial, and by any
-means.
+Copyright 2023 Parity Technologies (UK) Ltd.
```
Comment (ggwpez) [inline bin/node-template/LICENSE:3]:
```suggestion
Copyright Parity Technologies (UK) Ltd.
```
We don't use copyright years anymore - there is no gain and its annoying to maintain.

**Review by ggwpez — APPROVED** [review]

**Comment by ggwpez** [comment]
bot merge

**Pull request merged**